src/http.rs

```diff
@@ -137,18 +137,21 @@ fn submit_request(prep: RequestPrep) -> Result<String> {
         body: None,
     };
 
+    let headers = response.headers().clone();
+    for (key, value) in headers.iter() {
+        if let Ok(value) = value.to_str() {
+            resp.headers.insert(key.as_str(), value);
+        }
+    }
+
     if let Some(output_filename) = prep.output_filename {
-        std::io::copy(&mut response, &mut std::fs::File::create(&output_filename)?)?;
+        let mut writer = std::io::BufWriter::new(std::fs::File::create(&output_filename)?);
```

Comment:
`BufWriter`s are usually _much_ better in this situation, unbuffered I/O has a performance penalty.